examples/some-verifier/src/main.rs

```diff
@@ -287,213 +300,227 @@ async fn add_verification(
 
     let mut entry = VerificationsEntry::from_presentation(&proof);
     for proof in &proof.verifiable_credential {
-        proof_to_verifications_entry(state.clone(), proof, &mut entry)?;
+        state.proof_to_verifications_entry(proof, &mut entry)?;
     }
 
     match state.database.add_verification(entry).await {
-        Ok(()) => {}
-        // TODO: duplicate/overlapping entries
+        Ok(None) => {
+            tracing::info!("Successfully added new verification.");
+            Ok(StatusCode::CREATED)
+        }
+        Ok(Some(user_id)) => Err(Error::DuplicateUserIds(anyhow::anyhow!(
+            "Duplicate user id: {user_id}."
+        ))),
         Err(err) => {
             tracing::warn!("Error inserting entries: {err}");
-            return Err(Error::DatabaseError(err));
+            Err(Error::Database(err))
         }
     }
-
-    tracing::info!("Successfully verified.");
-
-    Ok(StatusCode::CREATED)
 }
 
 #[tracing::instrument(level = "info", skip_all)]
 async fn remove_verification(
     State(mut state): State<AppState>,
     request: Result<Json<Request>, JsonRejection>,
 ) -> Result<StatusCode, Error> {
-    tracing::info!("Request to remove verification.");
-
     let Json(request) = request?;
-    verify_request(&mut state, &request).await?;
-
-    let Request { proof, .. } = request;
+    let creds_with_metadata = state.verify_request(&request).await?;
 
     // Check the statements and add them to the database
-    let num_creds = proof.verifiable_credential.len();
-    if num_creds != 1 {
-        return Err(Error::NotSingleStatement(num_creds));
-    }
+    let Some((credential, &[])) = creds_with_metadata.credential_statements.split_first() else {
+        return Err(Error::NotSingleStatement(creds_with_metadata.credential_statements.len()));
+    };
```

Comment:
I am not sure I understand this comment.

This will match only if there is exactly one credential statement (the part `&[]` means the tail is empty.

Or are you referring to something else?

Reply:
I will merge this, but please respond if I have misunderstood your comment.

Reply:
I misunderstood what the code did, for some reason I thought that you just took nothing from the tail (but that doesn't make sense, I realize that 😛).